src/coreclr/src/gc/gc.cpp

```diff
@@ -23986,7 +23915,7 @@ void gc_heap::plan_phase (int condemned_gen_number)
 #endif //FEATURE_LOH_COMPACTION
         {
             GCToEEInterface::DiagWalkLOHSurvivors(__this);
-            sweep_large_objects();
+            sweep_uoh_objects (loh_generation);
```

Comment:
because loh is special here. it sweeps when not compacted. `sweep_uoh_objects (poh_generation)` will be called separately from this and unconditionally.

---
In reply to: [362299349](https://github.com/dotnet/runtime/pull/235#discussion_r362299349) [](ancestors = 362299349)